### ISSUE TITLE
[meta] updating old tools / python version pinning by default

### ISSUE THREAD

**Issue by hexylena** [galaxyproject/tools-iuc#6488]
hamronize: summarize version 1.0.3 uses `iteritems` meaning it is incompatible with python 3. I don't notice any issues running `hamronize` with py3 but the summarisation specifically is an issue.

This was fixed 2 years ago upstream: https://github.com/pha4ge/hAMRonization/commit/e560d836407d202f5a3754b3ea8a73eb9c73cc66  and github suggests it is included in v1.1.0 to v1.1.7 (latest), and IUC has an updated version of this.

*however* installing the older version which is used in workflows doesn't work specifically for one code path (html report.)

It is one of those cases where it would be nice if we could update old tools more easily. Add the dependency on py2 to ensure the conda environment gets resolved correctly.

_Simultaneously_, it'd be nice if IUC by policy, consider adding a current version of python to python tools, to prevent this situation in the future. Sure update the python version as we go along, but pinning the python version to something stable so that if a tool works today, it continues to work in the future when folks install it. Otherwise it's not reproducible without knowing which version of whatever language dependency might've been available when the tool was released, and manually fixing the conda environment on disk.

as a result I'm having to replace the version of hamronize in this workflow which has been well tested by others, I don't know that this new version doesn't change behaviours, but, I have no choice because I cannot easily tell consumers of the workflow that they will need to run some conda commands on their galaxy server to fix the dependencies.

**Comment by bgruening** [comment]
I'm not sure I completely understand the issue. Are you saying version 1.0.3 should have a dependency on python=2.7 and not 3.x? In this case this PR here is wrong https://github.com/bioconda/bioconda-recipes/pull/26528/files and we can create a new conda package that fixes those dependencies.

**Comment by hexylena** [comment]
Yes! correct. Version 1.0.3 of hamronize: summarize requires python 2.7. *Most* of the code paths are OK with py3 which I guess is why it wasn't caught. But specifically the html report generation fails because it still uses iteritems.
